### PR TITLE
[Storage] Fixed bug were BlobBaseClient.Exists() did not function correctly for blobs encrypted with CPK

### LINKED ISSUE #20361: [BUG] BlobClient.Exists() fails if a blob was uploaded with a customer-provided encryption key
**Describe the bug**
We have blobs that are uploaded with a `CustomerProvidedKey` (the REST equivalent would be uploading with the `x-ms-encryption-key` header).

At some later time, we want to check whether one of these blobs exists using the `BlobClient.Exists()` method. Importantly, the `BlobClient` instance used to call `Exists()` is *NOT* configured with a `CustomerProvidedKey`.

**Expected behavior**
Regardless of whether the `BlobClient` knows a blob's `CustomerProvidedKey`, the `Exists()` method should return `true` if the blob exists. (This is possible, see *Further Discussion* below.)

**Actual behavior**
When a `BlobClient` with no `CustomerProvidedKey` is used to call `Exists()` on a blob which was uploaded previously with a `CustomerProvidedKey`, it gets a 409 error response because the Azure Blob API expects the same encryption key to be passed.

The message returned is "The blob is encrypted with customer specified encryption, but it was not provided in the request", and the `x-ms-error-code` response header is set to "BlobUsesCustomerSpecifiedEncryption".

**Further Discussion**
I'm not disputing how the REST API works here; the fact that it's returning this error response makes perfect sense to me. The issue I have is with the SDK's handling of it: I'm just calling `BlobClient.Exists()`, and this particular error response from the REST API is proof positive that the blob does indeed exist...so my expectation is that the SDK would eat this particular error response and just return `true`, not error out.

**To Reproduce**
1. Upload a blob using a `CustomerProvidedKey`.
2. Using a new `BlobClient` instance pointed at said blob (which does not have a `CustomerProvidedKey` configured), call `Exists()`

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Blobs 12.8.1

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/20361

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 1]
```diff
@@ -3403,6 +3403,11 @@ private void SetNameFieldsIfNull()
                 {
                     return Response.FromValue(false, default);
                 }
+                catch (RequestFailedException storageRequestFailedException)
+                when (storageRequestFailedException.ErrorCode == BlobErrorCode.BlobUsesCustomerSpecifiedEncryption)
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs:3407]: @pakrym @tg-msft This is another example where we'd like to classify error as "info" for the telemetry purposes.
I think centralized place like "response classifier" might not scale to capture corner cases like this. Would it be possible for generated code to accept enumerable of "expected errors" or way to provide local response classifier?

File: sdk/storage/Azure.Storage.Blobs/src/Models/Internal/BlobImmutabilityPolicyMode.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage.Blobs.Models
+{
+    internal enum BlobImmutabilityPolicyMode
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Models/Internal/BlobImmutabilityPolicyMode.cs:6]: that doesn't seem to be related to PR.

File: sdk/storage/Azure.Storage.Blobs/src/Models/Internal/ListBlobsIncludeItemExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Storage.Blobs.Models
+{
+    internal partial class ListBlobsIncludeItemExtensions
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Models/Internal/ListBlobsIncludeItemExtensions.cs:8]: that doesn't seem to be related to PR.

**Review by kasobol-msft — COMMENTED** [review]
LGTM.
There's bunch of changes in the PR that don't seem to be related to the bug. Could you please remove them from PR?

**Comment by seanmcc-msft** [comment]
> LGTM.
> There's bunch of changes in the PR that don't seem to be related to the bug. Could you please remove them from PR?

I am unable to.  They are because I re-generated from swagger, which includes version level worm.  Not that basically nothing changed in sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs

File: sdk/storage/Azure.Storage.Blobs/src/Models/Internal/ListBlobsIncludeItemExtensions.cs (hunk 3 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/Internal/ListBlobsIncludeItemExtensions.cs:8]: Because I regenerated from swagger.

File: sdk/storage/Azure.Storage.Blobs/src/Models/Internal/BlobImmutabilityPolicyMode.cs (hunk 2 above)
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Models/Internal/BlobImmutabilityPolicyMode.cs:6]: Because I regenerated from swagger.

**Comment by kasobol-msft** [comment]
> > LGTM.
> > There's bunch of changes in the PR that don't seem to be related to the bug. Could you please remove them from PR?
> 
> I am unable to. They are because I re-generated from swagger, which includes version level worm. Note that basically nothing changed in sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs

To take new error code I suppose.

**Review by kasobol-msft — APPROVED** [review]

**Comment by seanmcc-msft** [comment]
> > > LGTM.
> > > There's bunch of changes in the PR that don't seem to be related to the bug. Could you please remove them from PR?
> > 
> > 
> > I am unable to. They are because I re-generated from swagger, which includes version level worm. Note that basically nothing changed in sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs
> 
> To take new error code I suppose.

Ya I wasn't happy with all the changes when I regenerated, either.  Sorry about that :(
